app/models/user.rb

```diff
@@ -15,6 +16,39 @@ class User < ApplicationRecord
                        length: {minimum: Settings.type_validate.min_length_password}, if: :password
   has_secure_password
 
+  class << self
+    def digest string
+      cost = if ActiveModel::SecurePassword.min_cost
+               BCrypt::Engine::MIN_COST
+             else
+               BCrypt::Engine.cost
+             end
+      BCrypt::Password.create string, cost: cost
+    end
+
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update remember_digest: User.digest(remember_token)
+  end
+
+  def authenticated? attribute, remember_token
+    digest = send("#{attribute}_digest")
+    if(digest.blank? || !remember_token)
```

Comment:
code chỗ này lạ vậy e
```suggestion
    return if digest.blank? || !remember_token
```
là dc.